### ISSUE TITLE
Why do I have to specify 'experimental' on appsody init?

### ISSUE THREAD

**Issue by ddunn2** [appsody/appsody#612]
## Summary
I noticed on the Appsody homepage that some of the `appsody init` commands I have to specify `experimental` whereas others I do not have to specify a repo. 

For example: 
`appsody init experimental/rocket`
`appsody init nodejs`

Why do I have to specify `experimental` for Rocket? I would have thought I could do: 
`appsody init rocket`

and it'd give me whatever the most stable option that is available. Then if it's giving me a non-stable stack then I would also expect to be told that within the command prompt. 

I feel I should only have to provide the `[repository]` if I'm wanting to use something specific. 

It'd also look better if the commands were all similar, I like the shortness of `appsody init <stack-name>`

Just something I noticed, and figured I'd raise here for discussion.

**Comment by skoh7645** [comment]
@ddunn2 if you run `appsody list` you can see a list of your available stacks. Under the repo column, the repo with an `*` in front of it is your default repo, so you don't need to specify the repo when running `appsody init`. The initial default repo users will get is the `incubator` repo. Users can change this using the command `appsody repo set-default <repo>`

**Comment by ddunn2** [comment]
@skoh7645 Thanks for the reply! 

I guess the point I'm making is why isn't `Appsody` my default repo? So when I do `appsody init rocket` it will just get me what is available? As currently, there isn't an `incubator` for `Rocket` for example. 

As I understand it there can't be both an `incubator` and an `experimental` version of a stack within Appsody right? So when I do `appsody init rocket` why can't Appsody figure out what is available and give me what currently exists? Rather than defaulting to `incubator` which may or may not exist.

**Comment by skoh7645** [comment]
Yes, in the case of the incubator and experimental repos there would never be a `rocket` stack in both repos as we manage those repos. But there could be multiple stacks across different repos with the same name, this is because users can add their own repos. In that case the CLI wouldn't know which to find

**Comment by ddunn2** [comment]
I would have thought `init` would default to the official Appsody Stacks unless configured otherwise. So if I do `appsody init rocket` I would get whatever version of the official Appsody stack is available by default. 

However if I wanted to use a different repo I can either provide it as `<repo>/<stackname>` or use `appsody repo set-default <repo>`  and then `init` would use the new default. 

This is just the behaviour I expected and was just curious as to why it works the way it does :).

**Comment by skoh7645** [comment]
@ddunn2 that's actually a good point... I'll mention this on our slack channel

**Comment by kylegc** [comment]
Slack thread: https://appsody.slack.com/archives/CKXDDB4A3/p1573226860025700

**Comment by kylegc** [comment]
I agree this would be nice to have. It is very similar to my original proposal https://github.com/appsody/stacks/issues/118#issuecomment-515159132

**Comment by neeraj-laad** [comment]
I agree it would be simpler if the repository name is only needed if there are multiple stacks with the same id across various repositories in the user's configuration.

**Comment by chilanti** [comment]
One of the reasons why it is done that way is for expedience. If we decide to go as suggested (i.e. provision a stack from any of the available repos, provided that there are no dups) there are two consequences:
1) Appsody will have to scan all repos for dups every time there's a non-fully qualified `init`. As far as I remember, this requires downloading all the indices, i.e. multiple HTTP requests. Now, we only download the index of the repository in question (i.e. the default repo in case of a non-fully qualified <stack>). 
2) The following scenario will have to be taken into account: say I create a script that performs `appsody init <stack>`. That script works successfully, until I add another repo, which also has <stack>. At which point my script may start failing.

**Comment by ddunn2** [comment]
Thanks for your input @chilanti 

So if I understand you correctly, `incubator`, `experimental` & `stable` are all separate repos? 
If that's the case, why aren't they just one repo? The official Appsody repo? Which contains stacks labeled with the correct 'stack level'? 

On your point `2.`, doesn't `appsody init <stack>` use whatever repo you've got set as default? So even if you added a new repo that also contains `<stack>` wouldn't `appsody init` still use the original repo unless you: 
1. Used: `appsody init <repo>/<stack>`
2. Set a new default repo: `appsody repo set-default <repo>`

I can only comment on this from a high level view though, as I don't know how Appsody works at its core :) 

I just found the current `appsody init` experience a little confusing and unexpected.
